### ISSUE TITLE
Sonarr 3.0.6.1342 - Certificate validation errors after updating to Mono 5.20.1.34-18

### ISSUE THREAD

**Issue by mreid-tt** [SynoCommunity/spksrc#5051]
### Setup
_Package Name:_ Sonarr
_Package Version:_ 20210717-19

_NAS Model:_ DS916+
_NAS Architecture:_ INTEL Pentium N3710
_DSM version:_ DSM 7.0.1-42218

### Expected behavior
Following the update of Mono to version 5.20.1.34-18, Sonarr is restarted. Once restarted, I should be able to check for updates without error from System -> Updates

### Actual behavior
On checking for updates I only get an error: Failed to fetch updates
I also note in the System -> Events a number of X509CertificateValidationService and SonarrErrorPipeline errors

### Steps to reproduce
_1._ Update Mono to latest version 5.20.1.34-18
_2._ Sonarr reboots as a dependent package
_3._ Errors show in the System -> Events

#### Package log
_Installer log: find logfile `/var/log/packages/{package}.log`, for DSM<6: `/var/packages/{package}/target/var/{package}_installer.log` and protocol in Package Center.
You find service log and other log files in `/var/packages/{package}/target/var`._

> nzbdrone_install.log -- https://pastebin.com/5EzrSXca
> nzbdrone.log -- https://pastebin.com/QjVGn1mu

#### Other logs
_E.g. `/var/log/messages` or `/var/log/synopkg.log`_

> synopkg.log -- https://pastebin.com/75sn1W1y

### Known workaround (see: https://github.com/SynoCommunity/spksrc/issues/5051#issuecomment-1009758625)

1. Within DSM, enable SSH service in _Control Panel > Terminal & SNMP_ and click apply
2. Using [Terminal](https://support.apple.com/en-gb/guide/terminal/apd5265185d-f365-44cb-8b09-71a064a42125/mac) (MacOS) connect to the NAS using `ssh -l [admin username] [NAS address]` or using [Putty](https://www.putty.org/) (Windows) connect to the network address of your NAS
3. Enter the required admin password and press enter
4. Enter the following command: `sudo /var/packages/mono/target/bin/cert-sync /etc/ssl/certs/ca-certificates.crt` and press enter
5. Enter the required admin password and press enter. When complete you should see the line _Import process completed_
6. Enter the following command: `sudo chmod -R a+rX /usr/share/.mono` and press enter
7. Disconnect the SSH session by typing `exit` and press enter
8. Within DSM, disable the SSH service in _Control Panel > Terminal & SNMP_ and click apply

**Comment by bakerboy448** [comment]
Not a bug - you need to sync mono's certs

https://community.synology.com/enu/forum/1/post/148065

**Comment by hgy59** [comment]
a regular installation and update run the following command in `service_postinst`:
```
${SYNOPKG_PKGDEST}/bin/cert-sync /etc/ssl/certs/ca-certificates.crt
```


this is shown in the installer log file like this (upgrade with already uptodate list)
```
2022/01/04 21:14:28     ===> Step postinst. USER= GROUP= SHARE_PATH=
2022/01/04 21:14:28     Begin save_wizard_variables
2022/01/04 21:14:28     End save_wizard_variables
2022/01/04 21:14:28     Begin service_postinst
Mono Certificate Store Sync - version 5.8.0.108
Populate Mono certificate store from a concatenated list of certificates.
Copyright 2002, 2003 Motus Technologies. Copyright 2004-2008 Novell. BSD licensed.

Importing into legacy system store:
I already trust 150, your new list has 150
Import process completed.

Importing into BTLS system store:
I already trust 150, your new list has 150
Import process completed.
2022/01/04 21:14:31     End service_postinst
2022/01/04 21:14:31     upgrade mono 5.8.0.108-11 End postinst ret=[0]
2022/01/04 21:14:31     upgrade mono 5.8.0.108-11 Begin postupgrade
2022/01/04 21:14:31     Begin reload_inst_variables
2022/01/04 21:14:31     End reload_inst_variables
2022/01/04 21:14:31     Begin initialize_variables
2022/01/04 21:14:31     End initialize_variables
2022/01/04 21:14:31     ===> Step postupgrade. USER= GROUP= SHARE_PATH=
2022/01/04 21:14:31     upgrade mono 5.8.0.108-11 End postupgrade ret=[0]
```


If you want to execute this command manually use /var/packages/mono/target/  for SYNOPKG_PKGDEST, i.e. run
`/var/packages/mono/target/bin/cert-sync /etc/ssl/certs/ca-certificates.crt`

**Comment by bakerboy448** [comment]
mono 5.8 really isn't supported

mono 5.18 or 5.20 are needed

**Comment by hgy59** [comment]
@mreid-tt analyzing your log file above, i am missing the `service_postinst` logs for mono.
probably this is an issue with automatic mono update when installing another package (nzbdrone/sonarr).

Probably you can resolve this, by manually installing (updating) the downloaded mono spk with identical version (5.20.1.34-18).

**Comment by hgy59** [comment]
> mono 5.8 really isn't supported
> 
> mono 5.18 or 5.20 are needed

I know, I know, this is only a sample installation log, that does not depend on mono version.

**Comment by hgy59** [comment]
It seems to be an issue on DSM 7

tried the same with mono 5.20.1.34-18 on DS218+ with DSM 7.0.1

`2022/01/04 21:45:35     System.UnauthorizedAccessException: Access to the path "/usr/share/.mono" is denied.`

here an extract of `/var/log/packages/mono.log`:

```
022/01/04 21:45:35     ===> Step postinst. USER= GROUP= SHARE_PATH=
2022/01/04 21:45:35     Begin save_wizard_variables
2022/01/04 21:45:35     End save_wizard_variables
2022/01/04 21:45:35     Begin syno_sync_var_folder
2022/01/04 21:45:35     End syno_sync_var_folder
2022/01/04 21:45:35     Begin service_postinst
2022/01/04 21:45:35     Mono Certificate Store Sync - version 5.20.1.34
2022/01/04 21:45:35     Populate Mono certificate store from a concatenated list of certificates.
2022/01/04 21:45:35     Copyright 2002, 2003 Motus Technologies. Copyright 2004-2008 Novell. BSD licensed.
2022/01/04 21:45:35     Importing into legacy system store:
2022/01/04 21:45:35     I already trust 0, your new list has 138
2022/01/04 21:45:35     Warning: Could not import CN=ACCVRAIZ1, OU=PKIACCV, O=ACCV, C=ES
2022/01/04 21:45:35     System.UnauthorizedAccessException: Access to the path "/usr/share/.mono" is denied.
2022/01/04 21:45:35       at System.IO.Directory.CreateDirectoriesInternal (System.String path) [0x0005e] in <a7386717e97e435ea6394b4446fbe4ba>:0
2022/01/04 21:45:35       at System.IO.Directory.CreateDirectory (System.String path) [0x0008f] in <a7386717e97e435ea6394b4446fbe4ba>:0
2022/01/04 21:45:35       at System.IO.DirectoryInfo.Create () [0x00000] in <a7386717e97e435ea6394b4446fbe4ba>:0
2022/01/04 21:45:35       at (wrapper remoting-invoke-with-check) System.IO.DirectoryInfo.Create()
2022/01/04 21:45:35       at System.IO.Directory.CreateDirectoriesInternal (System.String path) [0x00036] in <a7386717e97e435ea6394b4446fbe4ba>:0
2022/01/04 21:45:35       at System.IO.Directory.CreateDirectory (System.String path) [0x0008f] in <a7386717e97e435ea6394b4446fbe4ba>:0
2022/01/04 21:45:35       at System.IO.DirectoryInfo.Create () [0x00000] in <a7386717e97e435ea6394b4446fbe4ba>:0
2022/01/04 21:45:35       at (wrapper remoting-invoke-with-check) System.IO.DirectoryInfo.Create()
2022/01/04 21:45:35       at System.IO.Directory.CreateDirectoriesInternal (System.String path) [0x00036] in <a7386717e97e435ea6394b4446fbe4ba>:0
2022/01/04 21:45:35       at System.IO.Directory.CreateDirectory (System.String path) [0x0008f] in <a7386717e97e435ea6394b4446fbe4ba>:0
2022/01/04 21:45:35       at Mono.Security.X509.X509Store.CheckStore (System.String path, System.Boolean throwException) [0x00020] in <0a5b7d3822b84f2a96c7f10697a24fe9>:0
2022/01/04 21:45:35       at Mono.Security.X509.X509Store.Import (Mono.Security.X509.X509Certificate certificate) [0x00000] in <0a5b7d3822b84f2a96c7f10697a24fe9>:0
2022/01/04 21:45:35       at Mono.Tools.CertSync.ImportToStore (Mono.Security.X509.X509CertificateCollection roots, Mono.Security.X509.X509Store store) [0x00050] in <a7d568c231944f48bd7618f2ee421b9       4>:0
```


PS: it is time to port sonarr to .net!

**Comment by mreid-tt** [comment]
Hey @hgy59, thanks so much for the analysis. This does seem to be a DSM7 issue since the suggestion below did not work:

> Probably you can resolve this, by manually installing (updating) the downloaded mono spk with identical version (5.20.1.34-18).

I was able to manually import update the certificates by putting a `sudo` before your command below:

> If you want to execute this command manually use /var/packages/mono/target/ for SYNOPKG_PKGDEST, i.e. run `/var/packages/mono/target/bin/cert-sync /etc/ssl/certs/ca-certificates.crt`

The output looked like this:

```
Mono Certificate Store Sync - version 5.20.1.34
Populate Mono certificate store from a concatenated list of certificates.
Copyright 2002, 2003 Motus Technologies. Copyright 2004-2008 Novell. BSD licensed.

Importing into legacy system store:
I already trust 0, your new list has 138
Certificate added: CN=ACCVRAIZ1, OU=PKIACCV, O=ACCV, C=ES
Certificate added: C=ES, O=FNMT-RCM, OU=AC RAIZ FNMT-RCM
Certificate added: C=IT, L=Milan, O=Actalis S.p.A./03358520967, CN=Actalis Authentication Root CA
Certificate added: C=SE, O=AddTrust AB, OU=AddTrust External TTP Network, CN=AddTrust External CA Root
Certificate added: C=US, O=AffirmTrust, CN=AffirmTrust Commercial
Certificate added: C=US, O=AffirmTrust, CN=AffirmTrust Networking
Certificate added: C=US, O=AffirmTrust, CN=AffirmTrust Premium
Certificate added: C=US, O=AffirmTrust, CN=AffirmTrust Premium ECC
Certificate added: C=US, O=Amazon, CN=Amazon Root CA 1
Certificate added: C=US, O=Amazon, CN=Amazon Root CA 2
Certificate added: C=US, O=Amazon, CN=Amazon Root CA 3
Certificate added: C=US, O=Amazon, CN=Amazon Root CA 4
Certificate added: CN=Atos TrustedRoot 2011, O=Atos, C=DE
Certificate added: C=ES, CN=Autoridad de Certificacion Firmaprofesional CIF A62634068
Certificate added: C=IE, O=Baltimore, OU=CyberTrust, CN=Baltimore CyberTrust Root
Certificate added: C=NO, O=Buypass AS-983163327, CN=Buypass Class 2 Root CA
Certificate added: C=NO, O=Buypass AS-983163327, CN=Buypass Class 3 Root CA
Certificate added: C=SK, L=Bratislava, O=Disig a.s., CN=CA Disig Root R2
Certificate added: C=CN, O=China Financial Certification Authority, CN=CFCA EV ROOT
Certificate added: C=GB, S=Greater Manchester, L=Salford, O=COMODO CA Limited, CN=COMODO Certification Authority
Certificate added: C=GB, S=Greater Manchester, L=Salford, O=COMODO CA Limited, CN=COMODO ECC Certification Authority
Certificate added: C=GB, S=Greater Manchester, L=Salford, O=COMODO CA Limited, CN=COMODO RSA Certification Authority
Certificate added: C=FR, O=Dhimyotis, CN=Certigna
Certificate added: C=FR, O=Dhimyotis, OU=0002 48146308100036, CN=Certigna Root CA
Certificate added: C=PL, O=Unizeto Technologies S.A., OU=Certum Certification Authority, CN=Certum Trusted Network CA
Certificate added: C=PL, O=Unizeto Technologies S.A., OU=Certum Certification Authority, CN=Certum Trusted Network CA 2
Certificate added: C=EU, L=Madrid (see current address at www.camerfirma.com/address), SERIALNUMBER=A82743287, O=AC Camerfirma S.A., CN=Chambers of Commerce Root - 2008
Certificate added: C=GB, S=Greater Manchester, L=Salford, O=Comodo CA Limited, CN=AAA Certificate Services
Certificate added: O="Cybertrust, Inc", CN=Cybertrust Global Root
Certificate added: C=DE, O=D-Trust GmbH, CN=D-TRUST Root Class 3 CA 2 2009
Certificate added: C=DE, O=D-Trust GmbH, CN=D-TRUST Root Class 3 CA 2 EV 2009
Certificate added: O=Digital Signature Trust Co., CN=DST Root CA X3
Certificate added: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Assured ID Root CA
Certificate added: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Assured ID Root G2
Certificate added: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Assured ID Root G3
Certificate added: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Global Root CA
Certificate added: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Global Root G2
Certificate added: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Global Root G3
Certificate added: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert High Assurance EV Root CA
Certificate added: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Trusted Root G4
Certificate added: C=TR, L=Ankara, O=E-Tura EBG Biliim Teknolojileri ve Hizmetleri A.., OU=E-Tugra Sertifikasyon Merkezi, CN=E-Tugra Certification Authority
Certificate added: C=ES, O=Agencia Catalana de Certificacio (NIF Q-0801176-I), OU=Serveis Publics de Certificacio, OU=Vegeu https://www.catcert.net/verarrel (c)03, OU=Jerarquia Entitats de Certificacio Catalanes, CN=EC-ACC
Certificate added: C=EE, O=AS Sertifitseerimiskeskus, CN=EE Certification Centre Root CA, E=pki@sk.ee
Certificate added: O=Entrust.net, OU=www.entrust.net/CPS_2048 incorp. by ref. (limits liab.), OU=(c) 1999 Entrust.net Limited, CN=Entrust.net Certification Authority (2048)
Certificate added: C=US, O="Entrust, Inc.", OU=www.entrust.net/CPS is incorporated by reference, OU="(c) 2006 Entrust, Inc.", CN=Entrust Root Certification Authority
Certificate added: C=US, O="Entrust, Inc.", OU=See www.entrust.net/legal-terms, OU="(c) 2012 Entrust, Inc. - for authorized use only", CN=Entrust Root Certification Authority - EC1
Certificate added: C=US, O="Entrust, Inc.", OU=See www.entrust.net/legal-terms, OU="(c) 2009 Entrust, Inc. - for authorized use only", CN=Entrust Root Certification Authority - G2
Certificate added: C=US, O="Entrust, Inc.", OU=See www.entrust.net/legal-terms, OU="(c) 2015 Entrust, Inc. - for authorized use only", CN=Entrust Root Certification Authority - G4
Certificate added: C=CN, O="GUANG DONG CERTIFICATE AUTHORITY CO.,LTD.", CN=GDCA TrustAUTH R5 ROOT
Certificate added: C=US, O=Google Trust Services LLC, CN=GTS Root R1
Certificate added: C=US, O=Google Trust Services LLC, CN=GTS Root R2
Certificate added: C=US, O=Google Trust Services LLC, CN=GTS Root R3
Certificate added: C=US, O=Google Trust Services LLC, CN=GTS Root R4
Certificate added: C=US, O=GeoTrust Inc., CN=GeoTrust Global CA
Certificate added: C=US, O=GeoTrust Inc., CN=GeoTrust Primary Certification Authority
Certificate added: C=US, O=GeoTrust Inc., OU=(c) 2007 GeoTrust Inc. - For authorized use only, CN=GeoTrust Primary Certification Authority - G2
Certificate added: C=US, O=GeoTrust Inc., OU=(c) 2008 GeoTrust Inc. - For authorized use only, CN=GeoTrust Primary Certification Authority - G3
Certificate added: C=US, O=GeoTrust Inc., CN=GeoTrust Universal CA
Certificate added: C=US, O=GeoTrust Inc., CN=GeoTrust Universal CA 2
Certificate added: OU=GlobalSign ECC Root CA - R4, O=GlobalSign, CN=GlobalSign
Certificate added: OU=GlobalSign ECC Root CA - R5, O=GlobalSign, CN=GlobalSign
Certificate added: C=BE, O=GlobalSign nv-sa, OU=Root CA, CN=GlobalSign Root CA
Certificate added: OU=GlobalSign Root CA - R2, O=GlobalSign, CN=GlobalSign
Certificate added: OU=GlobalSign Root CA - R3, O=GlobalSign, CN=GlobalSign
Certificate added: OU=GlobalSign Root CA - R6, O=GlobalSign, CN=GlobalSign
Certificate added: C=EU, L=Madrid (see current address at www.camerfirma.com/address), SERIALNUMBER=A82743287, O=AC Camerfirma S.A., CN=Global Chambersign Root - 2008
Certificate added: C=US, O="The Go Daddy Group, Inc.", OU=Go Daddy Class 2 Certification Authority
Certificate added: C=US, S=Arizona, L=Scottsdale, O="GoDaddy.com, Inc.", CN=Go Daddy Root Certificate Authority - G2
Certificate added: C=GR, L=Athens, O=Hellenic Academic and Research Institutions Cert. Authority, CN=Hellenic Academic and Research Institutions ECC RootCA 2015
Certificate added: C=GR, O=Hellenic Academic and Research Institutions Cert. Authority, CN=Hellenic Academic and Research Institutions RootCA 2011
Certificate added: C=GR, L=Athens, O=Hellenic Academic and Research Institutions Cert. Authority, CN=Hellenic Academic and Research Institutions RootCA 2015
Certificate added: C=HK, O=Hongkong Post, CN=Hongkong Post Root CA 1
Certificate added: C=HK, S=Hong Kong, L=Hong Kong, O=Hongkong Post, CN=Hongkong Post Root CA 3
Certificate added: C=US, O=Internet Security Research Group, CN=ISRG Root X1
Certificate added: C=US, O=IdenTrust, CN=IdenTrust Commercial Root CA 1
Certificate added: C=US, O=IdenTrust, CN=IdenTrust Public Sector Root CA 1
Certificate added: C=ES, O=IZENPE S.A., CN=Izenpe.com
Certificate added: C=LU, O=LuxTrust S.A., CN=LuxTrust Global Root 2
Certificate added: C=HU, L=Budapest, O=Microsec Ltd., CN=Microsec e-Szigno Root CA 2009, E=info@e-szigno.hu
Certificate added: C=HU, L=Budapest, O=NetLock Kft., OU=Tanúsítványkiadók (Certification Services), CN=NetLock Arany (Class Gold) Ftanúsítvány
Certificate added: C=US, O=Network Solutions L.L.C., CN=Network Solutions Certificate Authority
Certificate added: C=CH, O=WISeKey, OU=Copyright (c) 2005, OU=OISTE Foundation Endorsed, CN=OISTE WISeKey Global Root GA CA
Certificate added: C=CH, O=WISeKey, OU=OISTE Foundation Endorsed, CN=OISTE WISeKey Global Root GB CA
Certificate added: C=CH, O=WISeKey, OU=OISTE Foundation Endorsed, CN=OISTE WISeKey Global Root GC CA
Certificate added: C=BM, O=QuoVadis Limited, OU=Root Certification Authority, CN=QuoVadis Root Certification Authority
Certificate added: C=BM, O=QuoVadis Limited, CN=QuoVadis Root CA 1 G3
Certificate added: C=BM, O=QuoVadis Limited, CN=QuoVadis Root CA 2
Certificate added: C=BM, O=QuoVadis Limited, CN=QuoVadis Root CA 2 G3
Certificate added: C=BM, O=QuoVadis Limited, CN=QuoVadis Root CA 3
Certificate added: C=BM, O=QuoVadis Limited, CN=QuoVadis Root CA 3 G3
Certificate added: C=US, S=Texas, L=Houston, O=SSL Corporation, CN=SSL.com EV Root Certification Authority ECC
Certificate added: C=US, S=Texas, L=Houston, O=SSL Corporation, CN=SSL.com EV Root Certification Authority RSA R2
Certificate added: C=US, S=Texas, L=Houston, O=SSL Corporation, CN=SSL.com Root Certification Authority ECC
Certificate added: C=US, S=Texas, L=Houston, O=SSL Corporation, CN=SSL.com Root Certification Authority RSA
Certificate added: C=PL, O=Krajowa Izba Rozliczeniowa S.A., CN=SZAFIR ROOT CA2
Certificate added: C=JP, O="Japan Certification Services, Inc.", CN=SecureSign RootCA11
Certificate added: C=US, O=SecureTrust Corporation, CN=SecureTrust CA
Certificate added: C=US, O=SecureTrust Corporation, CN=Secure Global CA
Certificate added: C=JP, O="SECOM Trust Systems CO.,LTD.", OU=Security Communication RootCA2
Certificate added: C=JP, O=SECOM Trust.net, OU=Security Communication RootCA1
Certificate added: C=FI, O=Sonera, CN=Sonera Class2 CA
Certificate added: C=NL, O=Staat der Nederlanden, CN=Staat der Nederlanden EV Root CA
Certificate added: C=NL, O=Staat der Nederlanden, CN=Staat der Nederlanden Root CA - G2
Certificate added: C=NL, O=Staat der Nederlanden, CN=Staat der Nederlanden Root CA - G3
Certificate added: C=US, O="Starfield Technologies, Inc.", OU=Starfield Class 2 Certification Authority
Certificate added: C=US, S=Arizona, L=Scottsdale, O="Starfield Technologies, Inc.", CN=Starfield Root Certificate Authority - G2
Certificate added: C=US, S=Arizona, L=Scottsdale, O="Starfield Technologies, Inc.", CN=Starfield Services Root Certificate Authority - G2
Certificate added: C=CH, O=SwissSign AG, CN=SwissSign Gold CA - G2
Certificate added: C=CH, O=SwissSign AG, CN=SwissSign Silver CA - G2
Certificate added: C=DE, O=T-Systems Enterprise Services GmbH, OU=T-Systems Trust Center, CN=T-TeleSec GlobalRoot Class 2
Certificate added: C=DE, O=T-Systems Enterprise Services GmbH, OU=T-Systems Trust Center, CN=T-TeleSec GlobalRoot Class 3
Certificate added: C=TR, L=Gebze - Kocaeli, O=Turkiye Bilimsel ve Teknolojik Arastirma Kurumu - TUBITAK, OU=Kamu Sertifikasyon Merkezi - Kamu SM, CN=TUBITAK Kamu SM SSL Kok Sertifikasi - Surum 1
Certificate added: C=TW, O=TAIWAN-CA, OU=Root CA, CN=TWCA Global Root CA
Certificate added: C=TW, O=TAIWAN-CA, OU=Root CA, CN=TWCA Root Certification Authority
Certificate added: C=TW, O=Government Root Certification Authority
Certificate added: O=TeliaSonera, CN=TeliaSonera Root CA v1
Certificate added: C=PA, S=Panama, L=Panama City, O=TrustCor Systems S. de R.L., OU=TrustCor Certificate Authority, CN=TrustCor ECA-1
Certificate added: C=PA, S=Panama, L=Panama City, O=TrustCor Systems S. de R.L., OU=TrustCor Certificate Authority, CN=TrustCor RootCert CA-1
Certificate added: C=PA, S=Panama, L=Panama City, O=TrustCor Systems S. de R.L., OU=TrustCor Certificate Authority, CN=TrustCor RootCert CA-2
Certificate added: C=GB, O=Trustis Limited, OU=Trustis FPS Root CA
Certificate added: C=CN, O=UniTrust, CN=UCA Extended Validation Root
Certificate added: C=CN, O=UniTrust, CN=UCA Global G2 Root
Certificate added: C=US, S=New Jersey, L=Jersey City, O=The USERTRUST Network, CN=USERTrust ECC Certification Authority
Certificate added: C=US, S=New Jersey, L=Jersey City, O=The USERTRUST Network, CN=USERTrust RSA Certification Authority
Certificate added: C=US, O="VeriSign, Inc.", OU=VeriSign Trust Network, OU="(c) 2007 VeriSign, Inc. - For authorized use only", CN=VeriSign Class 3 Public Primary Certification Authority - G4
Certificate added: C=US, O="VeriSign, Inc.", OU=VeriSign Trust Network, OU="(c) 2006 VeriSign, Inc. - For authorized use only", CN=VeriSign Class 3 Public Primary Certification Authority - G5
Certificate added: C=US, O="VeriSign, Inc.", OU=VeriSign Trust Network, OU="(c) 2008 VeriSign, Inc. - For authorized use only", CN=VeriSign Universal Root Certification Authority
Certificate added: C=US, O="VeriSign, Inc.", OU=VeriSign Trust Network, OU="(c) 1999 VeriSign, Inc. - For authorized use only", CN=VeriSign Class 3 Public Primary Certification Authority - G3
Certificate added: C=US, OU=www.xrampsecurity.com, O=XRamp Security Services Inc, CN=XRamp Global Certification Authority
Certificate added: C=RO, O=certSIGN, OU=certSIGN ROOT CA
Certificate added: C=TW, O="Chunghwa Telecom Co., Ltd.", OU=ePKI Root Certification Authority
Certificate added: C=US, OU=emSign PKI, O=eMudhra Inc, CN=emSign ECC Root CA - C3
Certificate added: C=IN, OU=emSign PKI, O=eMudhra Technologies Limited, CN=emSign ECC Root CA - G3
Certificate added: C=US, OU=emSign PKI, O=eMudhra Inc, CN=emSign Root CA - C1
Certificate added: C=IN, OU=emSign PKI, O=eMudhra Technologies Limited, CN=emSign Root CA - G1
Certificate added: C=US, O="thawte, Inc.", OU=Certification Services Division, OU="(c) 2006 thawte, Inc. - For authorized use only", CN=thawte Primary Root CA
Certificate added: C=US, O="thawte, Inc.", OU="(c) 2007 thawte, Inc. - For authorized use only", CN=thawte Primary Root CA - G2
Certificate added: C=US, O="thawte, Inc.", OU=Certification Services Division, OU="(c) 2008 thawte, Inc. - For authorized use only", CN=thawte Primary Root CA - G3
138 new root certificates were added to your trust store.
Import process completed.

Importing into BTLS system store:
I already trust 0, your new list has 138
Certificate added: CN=ACCVRAIZ1, OU=PKIACCV, O=ACCV, C=ES
Certificate added: C=ES, O=FNMT-RCM, OU=AC RAIZ FNMT-RCM
Certificate added: C=IT, L=Milan, O=Actalis S.p.A./03358520967, CN=Actalis Authentication Root CA
Certificate added: C=SE, O=AddTrust AB, OU=AddTrust External TTP Network, CN=AddTrust External CA Root
Certificate added: C=US, O=AffirmTrust, CN=AffirmTrust Commercial
Certificate added: C=US, O=AffirmTrust, CN=AffirmTrust Networking
Certificate added: C=US, O=AffirmTrust, CN=AffirmTrust Premium
Certificate added: C=US, O=AffirmTrust, CN=AffirmTrust Premium ECC
Certificate added: C=US, O=Amazon, CN=Amazon Root CA 1
Certificate added: C=US, O=Amazon, CN=Amazon Root CA 2
Certificate added: C=US, O=Amazon, CN=Amazon Root CA 3
Certificate added: C=US, O=Amazon, CN=Amazon Root CA 4
Certificate added: CN=Atos TrustedRoot 2011, O=Atos, C=DE
Certificate added: C=ES, CN=Autoridad de Certificacion Firmaprofesional CIF A62634068
Certificate added: C=IE, O=Baltimore, OU=CyberTrust, CN=Baltimore CyberTrust Root
Certificate added: C=NO, O=Buypass AS-983163327, CN=Buypass Class 2 Root CA
Certificate added: C=NO, O=Buypass AS-983163327, CN=Buypass Class 3 Root CA
Certificate added: C=SK, L=Bratislava, O=Disig a.s., CN=CA Disig Root R2
Certificate added: C=CN, O=China Financial Certification Authority, CN=CFCA EV ROOT
Certificate added: C=GB, S=Greater Manchester, L=Salford, O=COMODO CA Limited, CN=COMODO Certification Authority
Certificate added: C=GB, S=Greater Manchester, L=Salford, O=COMODO CA Limited, CN=COMODO ECC Certification Authority
Certificate added: C=GB, S=Greater Manchester, L=Salford, O=COMODO CA Limited, CN=COMODO RSA Certification Authority
Certificate added: C=FR, O=Dhimyotis, CN=Certigna
Certificate added: C=FR, O=Dhimyotis, OU=0002 48146308100036, CN=Certigna Root CA
Certificate added: C=PL, O=Unizeto Technologies S.A., OU=Certum Certification Authority, CN=Certum Trusted Network CA
Certificate added: C=PL, O=Unizeto Technologies S.A., OU=Certum Certification Authority, CN=Certum Trusted Network CA 2
Certificate added: C=EU, L=Madrid (see current address at www.camerfirma.com/address), SERIALNUMBER=A82743287, O=AC Camerfirma S.A., CN=Chambers of Commerce Root - 2008
Certificate added: C=GB, S=Greater Manchester, L=Salford, O=Comodo CA Limited, CN=AAA Certificate Services
Certificate added: O="Cybertrust, Inc", CN=Cybertrust Global Root
Certificate added: C=DE, O=D-Trust GmbH, CN=D-TRUST Root Class 3 CA 2 2009
Certificate added: C=DE, O=D-Trust GmbH, CN=D-TRUST Root Class 3 CA 2 EV 2009
Certificate added: O=Digital Signature Trust Co., CN=DST Root CA X3
Certificate added: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Assured ID Root CA
Certificate added: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Assured ID Root G2
Certificate added: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Assured ID Root G3
Certificate added: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Global Root CA
Certificate added: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Global Root G2
Certificate added: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Global Root G3
Certificate added: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert High Assurance EV Root CA
Certificate added: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Trusted Root G4
Certificate added: C=TR, L=Ankara, O=E-Tura EBG Biliim Teknolojileri ve Hizmetleri A.., OU=E-Tugra Sertifikasyon Merkezi, CN=E-Tugra Certification Authority
Certificate added: C=ES, O=Agencia Catalana de Certificacio (NIF Q-0801176-I), OU=Serveis Publics de Certificacio, OU=Vegeu https://www.catcert.net/verarrel (c)03, OU=Jerarquia Entitats de Certificacio Catalanes, CN=EC-ACC
Certificate added: C=EE, O=AS Sertifitseerimiskeskus, CN=EE Certification Centre Root CA, E=pki@sk.ee
Certificate added: O=Entrust.net, OU=www.entrust.net/CPS_2048 incorp. by ref. (limits liab.), OU=(c) 1999 Entrust.net Limited, CN=Entrust.net Certification Authority (2048)
Certificate added: C=US, O="Entrust, Inc.", OU=www.entrust.net/CPS is incorporated by reference, OU="(c) 2006 Entrust, Inc.", CN=Entrust Root Certification Authority
Certificate added: C=US, O="Entrust, Inc.", OU=See www.entrust.net/legal-terms, OU="(c) 2012 Entrust, Inc. - for authorized use only", CN=Entrust Root Certification Authority - EC1
Certificate added: C=US, O="Entrust, Inc.", OU=See www.entrust.net/legal-terms, OU="(c) 2009 Entrust, Inc. - for authorized use only", CN=Entrust Root Certification Authority - G2
Certificate added: C=US, O="Entrust, Inc.", OU=See www.entrust.net/legal-terms, OU="(c) 2015 Entrust, Inc. - for authorized use only", CN=Entrust Root Certification Authority - G4
Certificate added: C=CN, O="GUANG DONG CERTIFICATE AUTHORITY CO.,LTD.", CN=GDCA TrustAUTH R5 ROOT
Certificate added: C=US, O=Google Trust Services LLC, CN=GTS Root R1
Certificate added: C=US, O=Google Trust Services LLC, CN=GTS Root R2
Certificate added: C=US, O=Google Trust Services LLC, CN=GTS Root R3
Certificate added: C=US, O=Google Trust Services LLC, CN=GTS Root R4
Certificate added: C=US, O=GeoTrust Inc., CN=GeoTrust Global CA
Certificate added: C=US, O=GeoTrust Inc., CN=GeoTrust Primary Certification Authority
Certificate added: C=US, O=GeoTrust Inc., OU=(c) 2007 GeoTrust Inc. - For authorized use only, CN=GeoTrust Primary Certification Authority - G2
Certificate added: C=US, O=GeoTrust Inc., OU=(c) 2008 GeoTrust Inc. - For authorized use only, CN=GeoTrust Primary Certification Authority - G3
Certificate added: C=US, O=GeoTrust Inc., CN=GeoTrust Universal CA
Certificate added: C=US, O=GeoTrust Inc., CN=GeoTrust Universal CA 2
Certificate added: OU=GlobalSign ECC Root CA - R4, O=GlobalSign, CN=GlobalSign
Certificate added: OU=GlobalSign ECC Root CA - R5, O=GlobalSign, CN=GlobalSign
Certificate added: C=BE, O=GlobalSign nv-sa, OU=Root CA, CN=GlobalSign Root CA
Certificate added: OU=GlobalSign Root CA - R2, O=GlobalSign, CN=GlobalSign
Certificate added: OU=GlobalSign Root CA - R3, O=GlobalSign, CN=GlobalSign
Certificate added: OU=GlobalSign Root CA - R6, O=GlobalSign, CN=GlobalSign
Certificate added: C=EU, L=Madrid (see current address at www.camerfirma.com/address), SERIALNUMBER=A82743287, O=AC Camerfirma S.A., CN=Global Chambersign Root - 2008
Certificate added: C=US, O="The Go Daddy Group, Inc.", OU=Go Daddy Class 2 Certification Authority
Certificate added: C=US, S=Arizona, L=Scottsdale, O="GoDaddy.com, Inc.", CN=Go Daddy Root Certificate Authority - G2
Certificate added: C=GR, L=Athens, O=Hellenic Academic and Research Institutions Cert. Authority, CN=Hellenic Academic and Research Institutions ECC RootCA 2015
Certificate added: C=GR, O=Hellenic Academic and Research Institutions Cert. Authority, CN=Hellenic Academic and Research Institutions RootCA 2011
Certificate added: C=GR, L=Athens, O=Hellenic Academic and Research Institutions Cert. Authority, CN=Hellenic Academic and Research Institutions RootCA 2015
Certificate added: C=HK, O=Hongkong Post, CN=Hongkong Post Root CA 1
Certificate added: C=HK, S=Hong Kong, L=Hong Kong, O=Hongkong Post, CN=Hongkong Post Root CA 3
Certificate added: C=US, O=Internet Security Research Group, CN=ISRG Root X1
Certificate added: C=US, O=IdenTrust, CN=IdenTrust Commercial Root CA 1
Certificate added: C=US, O=IdenTrust, CN=IdenTrust Public Sector Root CA 1
Certificate added: C=ES, O=IZENPE S.A., CN=Izenpe.com
Certificate added: C=LU, O=LuxTrust S.A., CN=LuxTrust Global Root 2
Certificate added: C=HU, L=Budapest, O=Microsec Ltd., CN=Microsec e-Szigno Root CA 2009, E=info@e-szigno.hu
Certificate added: C=HU, L=Budapest, O=NetLock Kft., OU=Tanúsítványkiadók (Certification Services), CN=NetLock Arany (Class Gold) Ftanúsítvány
Certificate added: C=US, O=Network Solutions L.L.C., CN=Network Solutions Certificate Authority
Certificate added: C=CH, O=WISeKey, OU=Copyright (c) 2005, OU=OISTE Foundation Endorsed, CN=OISTE WISeKey Global Root GA CA
Certificate added: C=CH, O=WISeKey, OU=OISTE Foundation Endorsed, CN=OISTE WISeKey Global Root GB CA
Certificate added: C=CH, O=WISeKey, OU=OISTE Foundation Endorsed, CN=OISTE WISeKey Global Root GC CA
Certificate added: C=BM, O=QuoVadis Limited, OU=Root Certification Authority, CN=QuoVadis Root Certification Authority
Certificate added: C=BM, O=QuoVadis Limited, CN=QuoVadis Root CA 1 G3
Certificate added: C=BM, O=QuoVadis Limited, CN=QuoVadis Root CA 2
Certificate added: C=BM, O=QuoVadis Limited, CN=QuoVadis Root CA 2 G3
Certificate added: C=BM, O=QuoVadis Limited, CN=QuoVadis Root CA 3
Certificate added: C=BM, O=QuoVadis Limited, CN=QuoVadis Root CA 3 G3
Certificate added: C=US, S=Texas, L=Houston, O=SSL Corporation, CN=SSL.com EV Root Certification Authority ECC
Certificate added: C=US, S=Texas, L=Houston, O=SSL Corporation, CN=SSL.com EV Root Certification Authority RSA R2
Certificate added: C=US, S=Texas, L=Houston, O=SSL Corporation, CN=SSL.com Root Certification Authority ECC
Certificate added: C=US, S=Texas, L=Houston, O=SSL Corporation, CN=SSL.com Root Certification Authority RSA
Certificate added: C=PL, O=Krajowa Izba Rozliczeniowa S.A., CN=SZAFIR ROOT CA2
Certificate added: C=JP, O="Japan Certification Services, Inc.", CN=SecureSign RootCA11
Certificate added: C=US, O=SecureTrust Corporation, CN=SecureTrust CA
Certificate added: C=US, O=SecureTrust Corporation, CN=Secure Global CA
Certificate added: C=JP, O="SECOM Trust Systems CO.,LTD.", OU=Security Communication RootCA2
Certificate added: C=JP, O=SECOM Trust.net, OU=Security Communication RootCA1
Certificate added: C=FI, O=Sonera, CN=Sonera Class2 CA
Certificate added: C=NL, O=Staat der Nederlanden, CN=Staat der Nederlanden EV Root CA
Certificate added: C=NL, O=Staat der Nederlanden, CN=Staat der Nederlanden Root CA - G2
Certificate added: C=NL, O=Staat der Nederlanden, CN=Staat der Nederlanden Root CA - G3
Certificate added: C=US, O="Starfield Technologies, Inc.", OU=Starfield Class 2 Certification Authority
Certificate added: C=US, S=Arizona, L=Scottsdale, O="Starfield Technologies, Inc.", CN=Starfield Root Certificate Authority - G2
Certificate added: C=US, S=Arizona, L=Scottsdale, O="Starfield Technologies, Inc.", CN=Starfield Services Root Certificate Authority - G2
Certificate added: C=CH, O=SwissSign AG, CN=SwissSign Gold CA - G2
Certificate added: C=CH, O=SwissSign AG, CN=SwissSign Silver CA - G2
Certificate added: C=DE, O=T-Systems Enterprise Services GmbH, OU=T-Systems Trust Center, CN=T-TeleSec GlobalRoot Class 2
Certificate added: C=DE, O=T-Systems Enterprise Services GmbH, OU=T-Systems Trust Center, CN=T-TeleSec GlobalRoot Class 3
Certificate added: C=TR, L=Gebze - Kocaeli, O=Turkiye Bilimsel ve Teknolojik Arastirma Kurumu - TUBITAK, OU=Kamu Sertifikasyon Merkezi - Kamu SM, CN=TUBITAK Kamu SM SSL Kok Sertifikasi - Surum 1
Certificate added: C=TW, O=TAIWAN-CA, OU=Root CA, CN=TWCA Global Root CA
Certificate added: C=TW, O=TAIWAN-CA, OU=Root CA, CN=TWCA Root Certification Authority
Certificate added: C=TW, O=Government Root Certification Authority
Certificate added: O=TeliaSonera, CN=TeliaSonera Root CA v1
Certificate added: C=PA, S=Panama, L=Panama City, O=TrustCor Systems S. de R.L., OU=TrustCor Certificate Authority, CN=TrustCor ECA-1
Certificate added: C=PA, S=Panama, L=Panama City, O=TrustCor Systems S. de R.L., OU=TrustCor Certificate Authority, CN=TrustCor RootCert CA-1
Certificate added: C=PA, S=Panama, L=Panama City, O=TrustCor Systems S. de R.L., OU=TrustCor Certificate Authority, CN=TrustCor RootCert CA-2
Certificate added: C=GB, O=Trustis Limited, OU=Trustis FPS Root CA
Certificate added: C=CN, O=UniTrust, CN=UCA Extended Validation Root
Certificate added: C=CN, O=UniTrust, CN=UCA Global G2 Root
Certificate added: C=US, S=New Jersey, L=Jersey City, O=The USERTRUST Network, CN=USERTrust ECC Certification Authority
Certificate added: C=US, S=New Jersey, L=Jersey City, O=The USERTRUST Network, CN=USERTrust RSA Certification Authority
Certificate added: C=US, O="VeriSign, Inc.", OU=VeriSign Trust Network, OU="(c) 2007 VeriSign, Inc. - For authorized use only", CN=VeriSign Class 3 Public Primary Certification Authority - G4
Certificate added: C=US, O="VeriSign, Inc.", OU=VeriSign Trust Network, OU="(c) 2006 VeriSign, Inc. - For authorized use only", CN=VeriSign Class 3 Public Primary Certification Authority - G5
Certificate added: C=US, O="VeriSign, Inc.", OU=VeriSign Trust Network, OU="(c) 2008 VeriSign, Inc. - For authorized use only", CN=VeriSign Universal Root Certification Authority
Certificate added: C=US, O="VeriSign, Inc.", OU=VeriSign Trust Network, OU="(c) 1999 VeriSign, Inc. - For authorized use only", CN=VeriSign Class 3 Public Primary Certification Authority - G3
Certificate added: C=US, OU=www.xrampsecurity.com, O=XRamp Security Services Inc, CN=XRamp Global Certification Authority
Certificate added: C=RO, O=certSIGN, OU=certSIGN ROOT CA
Certificate added: C=TW, O="Chunghwa Telecom Co., Ltd.", OU=ePKI Root Certification Authority
Certificate added: C=US, OU=emSign PKI, O=eMudhra Inc, CN=emSign ECC Root CA - C3
Certificate added: C=IN, OU=emSign PKI, O=eMudhra Technologies Limited, CN=emSign ECC Root CA - G3
Certificate added: C=US, OU=emSign PKI, O=eMudhra Inc, CN=emSign Root CA - C1
Certificate added: C=IN, OU=emSign PKI, O=eMudhra Technologies Limited, CN=emSign Root CA - G1
Certificate added: C=US, O="thawte, Inc.", OU=Certification Services Division, OU="(c) 2006 thawte, Inc. - For authorized use only", CN=thawte Primary Root CA
Certificate added: C=US, O="thawte, Inc.", OU="(c) 2007 thawte, Inc. - For authorized use only", CN=thawte Primary Root CA - G2
Certificate added: C=US, O="thawte, Inc.", OU=Certification Services Division, OU="(c) 2008 thawte, Inc. - For authorized use only", CN=thawte Primary Root CA - G3
138 new root certificates were added to your trust store.
Import process completed.
```

Once this was completed, Sonarr is once again fully functional.

> PS: it is time to port sonarr to .net!

I agree, but I think I recall one of the devs saying that Sonarr is unlikely to do that anytime soon.

**Comment by bakerboy448** [comment]
Sonarr will not be moving to .net anytime soon

Based on the time it took for v3 to replace v2 I would not expect Sonarr in .net for several years at best.  Maybe I'll be wrong 🤷‍♂️

Sonarr on .Net also already exists for internal testing.
https://github.com/Sonarr/Sonarr/tree/widowmaker


For this SSL issue it seems that SynoCommunity's mono package has a bug and fails to sync certs upon install.  This should be corrected as soon as it can given all the support issues it's causing.

**Comment by bakerboy448** [comment]
Any timeframe to fix this bug?

we're up to a lot of users on a daily basis posting about this bug with the SynoCommunity mono package.

**Comment by mreid-tt** [comment]
I guess I could summarise the workaround for the community to follow in the meantime:

1. Within DSM, enable SSH service in _Control Panel > Terminal & SNMP_ and click apply
2. Using [Terminal](https://support.apple.com/en-gb/guide/terminal/apd5265185d-f365-44cb-8b09-71a064a42125/mac) (MacOS) connect to the NAS using `ssh -l [admin username] [NAS address]` or using [Putty](https://www.putty.org/) (Windows) connect to the network address of your NAS
3. Enter the required admin password and press enter
4. Enter the following command: `sudo /var/packages/mono/target/bin/cert-sync /etc/ssl/certs/ca-certificates.crt` and press enter
5. Enter the required admin password and press enter. When complete you should see the line _Import process completed_
6. Disconnect the SSH session by typing `exit` and press enter
7. Within DSM, disable the SSH service in _Control Panel > Terminal & SNMP_ and click apply

Once complete the errors in Sonarr should disappear on their own in a few minutes.

**Comment by Stanzilla** [comment]
`sudo /volume1/@appstore/mono/bin/cert-sync /etc/ssl/certs/ca-certificates.crt` for DSM 7, I think. Did not fix it for me though.

**Comment by bakerboy448** [comment]
DSM 7.1 for those this isn't fixing by chance?

it seems DSM 7.1 is causing a lot of package issues then.

**Comment by Stanzilla** [comment]
> DSM 7.1 for those this isn't fixing by chance?
> 
> it seems DSM 7.1 is causing a lot of package issues then.

Yep, 7.1.

**Comment by Patrick010** [comment]
Another 7.1 victim here.

**Comment by kunude** [comment]
Mono 5.20.1.34-17

**Comment by Patrick010** [comment]
> Mono 5.20.1.34-17

I downgraded to this version and it works now. Maybe you could elaborate a bit more on your comments ;)

**Comment by Stanzilla** [comment]
I can confirm this fixed it for me as well, you can download older versions from https://synocommunity.com/package/mono

**Comment by angelo-melis** [comment]
I have the same issue on DSM7.1. I did downgrade mono to Mono 5.20.1.34-17, and ran the cert-sync command. But still the same problem. Could someone explain what they did to get it to work on 7.1?

**Comment by bakerboy448** [comment]
It generally seems DSM7.1 is not supported for fresh installs and mono's certs cannot be synced.

**Comment by Patrick010** [comment]
> I have the same issue on DSM7.1. I did downgrade mono to Mono 5.20.1.34-17, and ran the cert-sync command. But still the same problem. Could someone explain what they did to get it to work on 7.1?

I basically only uninstalled 5.20.1.34-18 and manually installed 17. Didnt run the cert-sync.

**Comment by angelo-melis** [comment]
Thanks for the response, bakerboy448 and Patrick010. I also did an uninstall and manually install, but still couldn't get it to work. I switched to Docker for now and got it working.

**Comment by mmdriley** [comment]
I fixed this by updating permissions for Mono's machine certificate store to allow users like `sc-nzbdrone` to read:

```bash
$ sudo chmod -R a+rX /usr/share/.mono
```

**Comment by mmdriley** [comment]
If that folder doesn't exist, you may need to run the `cert-sync` commands listed above to create it.

The local machine store is under `${CommonApplicationData}/.mono`, which on Linux is `/usr/share`. ([ref](https://github.com/mono/mono/blob/41d4eb187476f9f983a2a7ec6f851bac6cd5e44e/mcs/class/Mono.Security/Mono.Security.X509/X509StoreManager.cs#L71-L81))

**Comment by hackgrid** [comment]
I updated to DSM 7.1-42661 Update 2 just now and had the same problem!

After

`sudo /volume1/@appstore/mono/bin/cert-sync /etc/ssl/certs/ca-certificates.crt`

and 

`$ sudo chmod -R a+rX /usr/share/.mono`


it worked fine again, thank you very much! :-)

**Comment by Agoris010** [comment]
> I updated to DSM 7.1-42661 Update 2 just now and had the same problem!
> 
> After
> 
> `sudo /volume1/@appstore/mono/bin/cert-sync /etc/ssl/certs/ca-certificates.crt`
> 
> and
> 
> `$ sudo chmod -R a+rX /usr/share/.mono`
> 
> it worked fine again, thank you very much! :-)

Worked like a charme. Thx

**Comment by hackgrid** [comment]
I guess #5070 has to be completed and merged first...?

**Comment by mmdriley** [comment]
Right, the workaround in my comments above is only "simple" because we can run it as `root`. Fixing this for real requires some nontrivial changes (keeping compatibility/upgrades in mind) to match DSM7's permission model.

**Comment by tonyellow** [comment]
chmod -R a+rX /usr/share/.mono worked for me too!

**Comment by Version3Synology** [comment]
Thanks a lot, both commands : 

- /volume1/@appstore/mono/bin/cert-sync /etc/ssl/certs/ca-certificates.crt
- chmod -R a+rX /usr/share/.mono

did it for me.

Was wondering why Radarr was able to reach indexers but not Sonarr, this thread brought the answer.

**Comment by TheGianni** [comment]
> 
> I fixed this on DSM 7.1 by updating permissions for Mono's machine certificate store to allow users like `sc-nzbdrone` to read:
> 
> ```shell
> $ sudo chmod -R a+rX /usr/share/.mono
> ```


You're the man! I already thought I'd have to switch everything to docker... Thanks!

**Comment by nieroivan** [comment]
Worked for me as well! Thank you guys!

**Comment by NMe84** [comment]
> Right, the workaround in my comments above is only "simple" because we can run it as `root`. Fixing this for real requires some nontrivial changes (keeping compatibility/upgrades in mind) to match DSM7's permission model.

Is anyone actually looking into those nontrivial changes? Is there anything the community can do to help? A DSM update just triggered the Sonarr package to get updated again, which once again broke certificates. It's of course simple to fix whenever it breaks but at this point it's getting annoying.

**Comment by we-are-borg** [comment]
For me chmod -R a+rX /usr/share/.mono is not working “no such file or directory” i’m on DSM 7.1.1-42962 Update 2 has the location changed.

**Comment by hackgrid** [comment]
You need to execute both commands mentioned further up!
After the latest update I also needed to do cert-sync and chmod again, but it worked fine immediately after that! :-)

**Comment by johncameron3** [comment]
Update: this was an issue again today. The same fix worked (again).

**Comment by jocamero** [comment]
Update: this was an issue again today. The same fix worked (again).

**Comment by mreid-tt** [comment]
> @mreid-tt regarding the mono certificate (and mono DSM 7) issues. I propose to create and install a script that does what you summarized in [#5051 (comment)](https://github.com/SynoCommunity/spksrc/issues/5051#issuecomment-1009758625).
> 
> This script could then be run (as root) with a task in DSM. The task could be executed on demand (or at DSM boot-up) and will avoid the need of an ssh login.
> 
> I prefer to add such a script to the mono package (there is still a pending PR #4669)

@hgy59, continuing this discussion here... how would you add a script to run as root exactly? In my reading online I get the impression that you cannot run scripts as root in DSM 7 unless your package is signed by Synology. Is there an approach you had in mind for me to try?

**Comment by bakerboy448** [comment]
> This script could then be run (as root) with a task in DSM.

^ User would need to configure it

**Comment by mmdriley** [comment]
DSM7 made it hard to run things as `root`, and for good reasons. I don't think we're going to get anywhere good by building an experience that depends on updating files outside SynoCommunity packages.

I feel like two good options here are:
1. Sonarr manages its own cert store under a filesystem path to which its user has full access and convinces Mono to use it. Unfortunately doesn't seem to be easy on the Mono side, see https://github.com/mono/mono/issues/6388.
2. The SynoCommunity Mono package takes more responsibility for maintaining the cert store Mono uses, then Sonarr and other packages benefit. Interestingly it looks like an attempt was made at this in #3154 but... it's not obvious the Mono that Sonarr depends on is actually built with that patch? I'm sure there's a good explanation but I haven't dug in to find it.

Continuing with (2), it also looks like the Mono package already tries to update its certificate store, though only just after installation: https://github.com/SynoCommunity/spksrc/blob/12ebc6cbff80ef0913016b98e964f96105f0fc6f/spk/mono_58/src/service-setup.sh#L1-L5

If we go with the Mono-package-owns-its-cert-store path, we may want to change that command to run on postinstall *and* every service start.

**Comment by Stanzilla** [comment]
Obsolete with v4 and the move to .NET Core anyway though, no?

**Comment by bakerboy448** [comment]
No ETA as to when v4 will be officially stable.

V3 was in beta for several years

**Comment by mreid-tt** [comment]
Packaged Sonarr v4 beta (https://github.com/SynoCommunity/spksrc/pull/5524) is ready to go. Just waiting on the admins to do a final review so it can be published for those who are willing to test it.

**Comment by jasonmp85** [comment]
Is this really the same issue going back a whole year?

**Comment by mreid-tt** [comment]
Think we've tracked down the problem with Mono and made a new PR #5604 for it. Thus far testing seems to be working well. Until it is published, those willing can download from the [Artifacts](https://github.com/SynoCommunity/spksrc/actions/runs/4158324234) section of the build for testing.
